projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/templates/load/fact/insert/02-handle-quality-checks.py

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: Apache-2.0
+import os
+
+from vdk.api.job_input import IJobInput
+from vdk.plugin.impala.templates.utility import align_stg_table_with_target
+from vdk.plugin.impala.templates.utility import get_file_content
+from vdk.plugin.impala.templates.utility import get_staging_table_name
+
+SQL_FILES_FOLDER = (
+    os.path.dirname(os.path.abspath(__file__)) + "/02-requisite-sql-scripts"
+)
+
+"""
+This step is intened to handle quality checks if such provided
+and stop the data from being populated into the target table if the check has negative outcome.
+Otherwise the data will be directly processed according to the used template type
+"""
+
+
+def run(job_input: IJobInput):
+    job_arguments = job_input.get_arguments()
+
+    check = job_arguments.get("check")
+    partition_clause = job_arguments["_vdk_template_insert_partition_clause"]
+    source_schema = job_arguments.get("source_schema")
+    source_view = job_arguments.get("source_view")
+    target_schema = job_arguments.get("target_schema")
+    target_table = job_arguments.get("target_table")
+    insert_query = get_file_content(SQL_FILES_FOLDER, "02-insert-into-target.sql")
+
+    if check:
+        staging_schema = job_arguments.get("staging_schema", target_schema)
+        staging_table_name = get_staging_table_name(target_schema, target_table)
+
+        staging_table = f"{staging_schema}.{staging_table_name}"
+
+        align_stg_table_with_target(
+            f"{target_schema}.{target_table}", staging_table, job_input
+        )
+
+        insert_into_staging = insert_query.format(
+            target_schema=staging_schema,
+            target_table=staging_table_name,
+            _vdk_template_insert_partition_clause=partition_clause,
+            source_schema=source_schema,
+            source_view=source_view,
+        )
+        job_input.execute_query(insert_into_staging)
+
+        view_schema = staging_schema
+        view_name = f"vw_{staging_table_name}"
+        create_view_query = get_file_content(
+            SQL_FILES_FOLDER, "02-create-consolidated-view.sql"
+        )
+        create_view = create_view_query.format(
+            view_schema=view_schema,
+            view_name=view_name,
+            target_schema=target_schema,
+            target_table=target_table,
+            staging_schema=staging_schema,
+            staging_table_name=staging_table_name,
+        )
+        job_input.execute_query(create_view)
+
+        if check(f"{view_schema}.{view_name}"):
+            insert_into_target = insert_query.format(
+                source_schema=staging_schema,
+                source_view=staging_table_name,
+                _vdk_template_insert_partition_clause=partition_clause,
+                target_schema=target_schema,
+                target_table=target_table,
+            )
+            job_input.execute_query(insert_into_target)
+        else:
+            raise Exception("The data is not passing the quality checks!")
```

Comment:
Avoid raising base exception . This is called type of [error hiding](https://en.wikipedia.org/wiki/Error_hiding) that causes issues. 

You can either create a new exception type (I think having DataQualityException would be the best). In this exception class you can even add custom fields with more information about the data quality check (context) which might provide the user (likely you in the future) more information on what is going on.